### ISSUE TITLE
bug: No LSP client found that supports symbols when deleteing buffer

### ISSUE THREAD

**Issue by mehalter** [stevearc/aerial.nvim#393]
### Neovim version (nvim -v)

NVIM v0.10.1 (github release)

### Operating system/version

Arch Linux

### Output of :AerialInfo

Aerial Info                                                                                                                                                    
-----------                                                                                                                                                    
Filetype: lua                                                                                                                                                  
Configured backends:                                                                                                                                           
  lsp (supported) (attached)                                                                                                                                   
Show symbols: Class, Constructor, Enum, Function, Interface, Module, Method, Struct

### Describe the bug

When using Neovim v0.10 I am getting an error message, "No LSP client found that supports symbols in buffer X" where X is the buffer that I have just deleted. It seems like this is only happening on Neovim v0.10 and it does not happen on Neovim v0.9.

### What is the severity of this bug?

minor (annoyance)

### Steps To Reproduce

0. This sets up the Lua language server, so expects that `lua-language-server` is in the $PATH. If you want to test with another language server you will have to modify the `repro.lua` file to set it up and adjust the commands below.
1. `nvim -u repro.lua`, spin up the minimal reproducing configuration below
2. `:e repro.lua`, open up a lua file so that the language server attaches
3. `:e test.lua`, open up a new lua file so we have a second one
4. `:bdelete`, close our new lua file
5. See the notification at the bottom

### Expected Behavior

There should be no error notification.

### Minimal example file

N/A the `repro.lua` is enough

### Minimal init.lua

```Lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

-- install plugins
require("lazy.minit").repro({
  spec = {
    {
      "stevearc/aerial.nvim",
      config = function()
        require("aerial").setup({
          backends = { "lsp" }, -- prefer LSP
          on_attach = function(bufnr)
            -- this function purely just has to exist
          end,
        })
      end,
    },
    {
      "nvim-treesitter/nvim-treesitter",
      build = ":TSUpdate",
      config = function()
        require("nvim-treesitter.configs").setup({
          auto_install = true,
          highlight = { enable = true },
        })
      end,
    },
    {
      "neovim/nvim-lspconfig",
      config = function()
        require("lspconfig").lua_ls.setup({})
      end,
    },
  },
})
```

**Comment by mehalter** [comment]
Upon more investigation it looks like the requirement here is that there needs to be an `on_attach` function. If none is provided then the problem goes away

**Comment by chaozwn** [comment]
> Upon more investigation it looks like the requirement here is that there needs to be an `on_attach` function. If none is provided then the problem goes away

I've rolled back to neovim 0.10.0.

**Comment by mehalter** [comment]
I think I still get the bug in 0.10.0

**Comment by mehalter** [comment]
@chaozwn I can confirm that this bug is still reproducible in neovim 0.10.0

**Comment by mehalter** [comment]
I started digging into this. It looks like it's because of the scheduled call to `attach` when a language server detaches happening when a buffer is actively getting deleted:

https://github.com/stevearc/aerial.nvim/blob/master/lua/aerial/backends/lsp/init.lua#L129-L136

**Comment by rami3l** [comment]
> Upon more investigation it looks like the requirement here is that there needs to be an `on_attach` function. If none is provided then the problem goes away

@mehalter The same workaround has been reported in https://github.com/stevearc/aerial.nvim/issues/236#issuecomment-1529312757: 

> I find that if I commented out the on_attach specification in my config, then such warning when using :q! to quit a buffer will not happen

... but that issue has been closed in favor of #249, which is also closed. Is it coming back again?

**Comment by mehalter** [comment]
@rami3l yes, given the minimal reproducing configuration I have provided here it is still happening

**Comment by mehalter** [comment]
The real root of this issue is that when the `LspDetach` event is called the buffer has not yet been deleted, but will be deleted after the autocommands evaluate. So it always will say the buffer is valid while trying to fetch the symbols and therefore it shows that the there are no more language servers are attached. I think the ordering of these operations were probably incorrect in 0.9 and was fixed in 0.10 which led to this re-emerging. I'm honestly not sure the best approach for this outside of maybe just removing the warning that no language servers are attached when trying to fetch symbols 🤔 

What do you think @stevearc ? Is this message actually necessary? I think in the case where this matters the `is_supported` call probably handles this error automatically. From my testing it seems like we can simply remove that warning message altogether.
